### PR TITLE
update the process of loading genomics data

### PR DESCRIPTION
**Pull request by CunliangGeng**
Update loading process in the `DatasetLoader._load_genomics` function.

### CODE REVIEW

**Comment by CunliangGeng** [comment]
Current dependencies on/for this PR:
* `dev`
  * **PR #178** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/178?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #181** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/181?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #182** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/182?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #183** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/183?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #184** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/184?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #185** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/185?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
              * **PR #186** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/186?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #187** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/187?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #188** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/188?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #189** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/189?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #190** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/190?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #191** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/191?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #192** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/192?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #193** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/193?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                              * **PR #194** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/194?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #196** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/196?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                  * **PR #197** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/197?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                    * **PR #198** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/198?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                      * **PR #199** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/199?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                        * **PR #200** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/200?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                          * **PR #201** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/201?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                            * **PR #202** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/202?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                                * **PR #195** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/195?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/185?utm_source=stack-comment).

**Comment by CunliangGeng** [comment]
Current dependencies on/for this PR:
* dev
  * **PR #178** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/178?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #181** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/181?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #182** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/182?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #183** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/183?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #184** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/184?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #185** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/185?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/185?utm_source=stack-comment).

File: src/nplinker/loader.py [hunk 1]
```diff
@@ -439,22 +437,21 @@ def _load_genomics(self):
...
         map_strain_to_bgc(self.strains, raw_bgcs)
 
         # Step 4: assign BGC objects to GCF.bgcs
         map_bgc_to_gcf(raw_bgcs, raw_gcfs)
 
-        # Step 5: get clean GCF objects, BGC objects and Strain objects
+        # Step 5: get GCF objects and their BGC members
         self.gcfs = raw_gcfs
         self.bgcs = get_bgcs_from_gcfs(self.gcfs)
-        self.strains = get_strains_from_bgcs(self.bgcs)
```
Comment (gcroci2) [inline src/nplinker/loader.py:457]: Now `self.strains` is assigned in `_load_strain_mappings()`, correct?

**Review by gcroci2 — APPROVED** [review]

File: src/nplinker/loader.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/loader.py:457]: It's more complex than that. After loading from `strain mapping`, it's also needed to filter out user's specified strains and then get common strains from both sides (genomics and metabolomics). I'm just working on that, and you will see that in coming PRs.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Dec 14, 5:52 AM**: @@CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/185).
* **Dec 14, 5:54 AM**: [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/185) rebased this pull request as part of a merge.
* **Dec 14, 5:56 AM**: @@CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/185).

**Pull request merged**